File: app/src/main/java/com/goranch/publicapis/ui/food/viewmodel/IFoodViewModel.java
package com.goranch.publicapis.ui.food.viewmodel;

import com.goranch.publicapis.api.model.food.recipe.Recipe;

interface IFoodViewModel {
    void getRecipes(String query);

    void getSingleRecipe(String recipeId);

    void getNaturalLanguageNutritionInfo(String ingredients);

    void onItemClicked(Recipe item);
}

File: app/src/main/java/com/goranch/publicapis/ui/food/viewmodel/IFoodViewModel.kt
package com.goranch.publicapis.ui.food.viewmodel

import com.goranch.publicapis.api.model.food.recipe.Recipe

internal interface IFoodViewModel {
    fun getRecipes(query: String)

    fun getSingleRecipe(recipeId: String)

    fun getNaturalLanguageNutritionInfo(ingredients: String)

    fun onItemClicked(item: Recipe)
}
